### ISSUE TITLE
Rich Text Editor does not Insert Links with Proper Markedown [title](url...)

### ISSUE THREAD

**Issue by drankinatty** [nextcloud/notes#1134]
Please make a clear and concise description of what the bug is.

## Steps to reproduce
1. Create Note, choose Insert Link
2. Type URL

### Expected behaviour
Tell us what should happen

"Markdown" format for a link is `[text to display](URL...)` which will display `"text to display"` with hyperlink to `"URL..."`. When entering a link in NC Notes, the user should have the ability to insert `"text to display"` as well as the URL. For markdown link reference, see the Wikipedia link [Markdown](https://en.wikipedia.org/wiki/Markdown) that the Notes Readme.md references.

(and curiously, simply copying a link and pasting here in this new Issue on github uses the proper markdown syntax... for the link)  

### Actual behaviour
Tell us what happens instead

Instead of allowing input of `text to display` and inserting the link with proper markdown format, NC Notes places the URL between `< url... >` and displays only the raw URL followed by rectangular box containing the link at the page title (if available) in Rich Text Editor Mode, and displays only the raw URL in Preview mode.

### Screenshots

Attempting to enter a link to `https://www.mysite.com` with the *text to display* as `"mysite"` is impossible in Rich Text Mode. Instead you enter the link and are left with:

![nc-RTM-link](https://github.com/nextcloud/notes/assets/9997638/ea947ba2-a6ec-4d4a-b862-186fe7369fbe)

Attempting to enter a link in proper markdown format in Rich Text Mode fails, e.g.

![nc-attempt-to-enter-link](https://github.com/nextcloud/notes/assets/9997638/1c2df139-178b-4487-b868-9b8c02c41ebe)

(**note:** how NC crams the URL between `< ... >` inside the attempted proper markdown format and escapes the `\[ ... \]` of the proper markdown)

It's only after changing "Note Settings" to display in Preview/Edit mode that you can fix the incorrect markdown format inserted by NC Notes in Rich Text Mode, e.g.

![nc-manually-edit-link](https://github.com/nextcloud/notes/assets/9997638/d40767ae-5f8e-41af-ba88-b71374753b00)

Now, finally, the link displays properly in BOTH Preview and Rich Text Mode:

![nc-final-link](https://github.com/nextcloud/notes/assets/9997638/f5e16450-eeaa-47ec-bdc7-facaf3845192)

## Server
Please complete the following information.

- Notes app version:  4.8.1
- Nextcloud version:  27.1.2
- OS: Archlinux: 6.5.7-arch1-1
- Web server: Apache / 2.4.57 (Unix)
- PHP version: PHP / 8.1.23
- Database: mariadb / 11.1.2-1

**Nextcloud configuration:**

<details>
```
{
    "system": {
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "default_locale": "en_US",
        "default_phone_region": "US",
        "knowledgebaseenabled": true,
        "apps_paths": [
            {
                "path": "\/usr\/share\/webapps\/nextcloud\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/lib\/nextcloud\/apps",
                "url": "\/wapps",
                "writable": true
            }
        ],
        "trusted_domains": [
            "localhost",
            "2pi.mysite.com",
            "192.168.6.111"
        ],
        "overwrite.cli.url": "https:\/\/2pi.3111skyline.com\/nextcloud",
        "htaccess.RewriteBase": "\/nextcloud",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "27.1.2.1",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25",
        "maintenance": false,
        "app_install_overwrite": [
            "issuetemplate"
        ],
        "theme": "",
        "loglevel": 2,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        }
    }
}

```
</details>

## Client
Please complete the following information.

- Browser (incl. version): Firefox 115.3.1esr (64-bit)
 - OS: Archlinux

## Log files
N/A

**Comment by juliushaertl** [comment]
Thanks for reporting. 

The easiest way to add a title and link it is to type the title first and then mark it and either paste the link or insert it through the menu: 
<img width="755" alt="Screenshot 2024-06-07 at 13 05 42" src="https://github.com/nextcloud/notes/assets/3404133/6a9766d8-1f0e-489b-b602-813470ff2b23">

However I see that the issues you describe are still worth to address.

Link previews have been changed to be optional to enable case by case with Nextcloud 29, so you should no longer be left with a empty preview there (if the page does not provide opengraph data)

Inserting with typing markdown syntax is tracked in https://github.com/nextcloud/text/issues/55 so I'd close this one as the editor implementation is all part of the text app.
